### ISSUE TITLE
.NET 7 RC1 improvements to Rate limiting middleware

### ISSUE THREAD

**Issue by Rick-Anderson** [dotnet/AspNetCore.Docs.Samples#45]
[Rate limiting middleware improvements](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-7-rc-1/#rate-limiting-middleware-improvements)

* [Code to update](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/middleware/rate-limit/WebRateLimitAuth)
* Add `[EnableRateLimitingAttribute("MyControllerPolicy")]` to [`Home2Controller`](https://github.com/dotnet/AspNetCore.Docs.Samples/blob/main/fundamentals/middleware/rate-limit/WebRateLimitAuth/Controllers/Home2Controller.cs)
* disable rate limiting on a specific endpoint within that group
* apply policies directly to endpoints.
* add an instance of it directly to an endpoint
* Add 

 ```
builder.Services.AddRateLimiter(options =>
{
    options.AddTokenBucketLimiter("MyControllerPolicy", options =>
    {
        options.TokenLimit = 1;
        options.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
        options.QueueLimit = 1;
        options.ReplenishmentPeriod = TimeSpan.FromSeconds(10);
        options.TokensPerPeriod = 1;
    })
    .AddPolicy<string>("MyGroupPolicy", new MyRateLimiterPolicy());
});
```

Then @Rick-Anderson  can update [Rate limiting middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/rate-limit?view=aspnetcore-7.0)

@Kahbazi can you do these updates?

Also, the updates requested in #14

**Comment by Rick-Anderson** [comment]
@fiyazbinhasan @sammychinedu2ky  anyone want to update this sample code for RC1?

**Comment by Rick-Anderson** [comment]
@maartenba  would you be able to update the [Code ](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/middleware/rate-limit/WebRateLimitAuth). I'll put you on the authors list of [Rate limiting middleware in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/rate-limit?view=aspnetcore-7.0)

**Comment by sammychinedu2ky** [comment]
> @fiyazbinhasan @sammychinedu2ky anyone want to update this sample code for RC1?

sorry missed this

**Comment by maartenba** [comment]
Here you go folks - https://github.com/dotnet/AspNetCore.Docs.Samples/pull/82

**Comment by Rick-Anderson** [comment]
Fixed in #82
